mirrord/cli/src/verify_config.rs

```diff
@@ -56,6 +57,30 @@ impl From<TargetConfig> for VerifiedTargetConfig {
     }
 }
 
+/// Corresponds to variants of [`Target`].
+#[derive(Serialize, PartialEq, Eq, Clone, Copy)]
+#[serde(rename_all = "lowercase")]
+enum TargetType {
+    Targetless,
+    Pod,
+    Deployment,
+    Rollout,
+}
+
+impl TargetType {
+    fn all() -> impl Iterator<Item = Self> {
+        [Self::Targetless, Self::Pod, Self::Deployment, Self::Rollout].into_iter()
+    }
+
+    fn compatible_with(&self, config: &FeatureConfig) -> bool {
+        match self {
+            Self::Targetless | Self::Rollout => !config.copy_target.enabled,
+            Self::Pod => !(config.copy_target.enabled && config.copy_target.scale_down),
```

Comment:
Do we have these config matches documented anywhere (don't remember for the `copy_target` stuff)?

If we don't could you doc it saying why they're not allowed.

Reply:
Updated `copy_target` docs, now it mentions compatibility with target types :+1: